xclim/indices/_threshold.py

```diff
@@ -1237,6 +1238,62 @@ def days_with_snow(
     return to_agg_units(out, prsn, "count")
 
 
+@declare_units(tasmax="[temperature]", hurs="[g m-3]", thresh="[temperature]")
+def heat_index(
+    tasmax: xarray.DataArray,
+    hurs: xarray.DataArray,
+    thresh: str = "20.0 degC",
+    freq: str = "MS",
+) -> xarray.DataArray:
+    r"""Daily heat index.
+
+    Days in which the human body perceives temperature is hot, which combines
+    relative humidity with the air temperature.
+
+    Parameters
+    ----------
+    tasmax : xarray.DataArray
+      Maximum daily temperature.
+    hurs : xarray.DataArray
+      Relative humidity.
+    thresh : str
+      Threshold temperature on which to base evaluation.
+    freq : str
+      Resampling frequency.
+
+    Returns
+    -------
+    xarray.DataArray, [time][temperature]
+      Heat index.
+    """
+    thresh = convert_units_to(thresh, tasmax)
+    t = convert_units_to(tasmax, "degC")
+    t = t.where(t > 20)
+    r = convert_units_to(hurs, "%")
+
+    tr = np.multiply(t, r)
+    tt = np.multiply(t, t)
+    rr = np.multiply(r, r)
+    ttr = np.multiply(tt, r)
+    trr = np.multiply(t, rr)
+    ttrr = np.multiply(tt, rr)
+
+    out = (
+        -8.78469475556
+        + 1.61139411 * t
+        + 2.33854883889 * r
+        - 0.14611605 * tr
+        - 0.012308094 * tt
+        - 0.0164248277778 * rr
+        + 0.002211732 * ttr
+        + 0.00072546 * trr
+        - 0.000003582 * ttrr
+    )
```

Comment:
I'm guessing there is a source for these factors? Please add it to the docstring in a "References" section.

Reply:
@aulemahal unfortunately the only source of these *specific* coefficients are from [the wikipedia article on the heat index](https://en.wikipedia.org/wiki/Heat_index). The references within all give formulae in Fahrenheit, not Celsius and the coefficients seem to have been converted for the wiki article. We could rewrite the formula in F, converting units in/out of the function from/to Celsius, allowing us to directly reference the proper source, or we can just reference the wikipedia article. What are your thoughts?

Reply:
I see! I would cite at least one reference and mention that factors have been converted to Celsius. Or indeed, doing this all in Fahrenheit could be even more explicit, but I think we want to have the output Celsius-like so it may be too much noise.

I see that there are more than one formulation! Something we do in xclim (see relative humidity for example) is to have a `method` arg which is a string to switch between different implementations. It would make sense here as it is mostly only the coefficients that change. But no need to do this here, it can be done later when the need arises.

Reply:
@aulemahal Sorry, I didn't get a notification about this conversation... the reference for these factors is [blazejczyk2012] Blazejczyk, K., Epstein, Y., Jendritzky, G., Staiger, H., & Tinz, B. (2012). Comparison of UTCI to selected thermal indices. International journal of biometeorology, 56(3), 515-535 (I think that @raquel-ucl may have added this already?)

Reply:
Yes, I added it in https://github.com/Ouranosinc/xclim/pull/915/commits/6327f32ccf0cba9e12e596e839c5d8deac39781e